### DISCUSSION TITLE
Customized API based extensions will timeout when running for more than one minute

### DISCUSSION THREAD

**Discussion by Arvee-Li** [langgenius/dify#4749] [General]
I deployed an audio to text tool (FunASR) locally and successfully integrated the model into Dify through FastAPI. But when the audio to be translated is long (such as 2 hours), FunASR can work successfully locally, but the tool node in Dify will report an error due to timeout. So how can I adjust the response time of the tool nodes in Dify to allow the tool to run for a long time without reporting errors?

![1ecec1fbbcd1b7c6c8f1019737f8bf5](https://github.com/langgenius/dify/assets/59771623/085e2c3f-3577-467e-a4a3-e1145906121b)

![db6cf7ec543b12561722cffa8c1188b](https://github.com/langgenius/dify/assets/59771623/e0b0381a-29c7-419d-a6ae-ead2d91d8773)

![73d2993e130d95b9af204258996a8af](https://github.com/langgenius/dify/assets/59771623/7d0d6f59-23a5-4601-b810-024ab2ed553b)

**Comment by justemu** [comment]
It seems that you wrote a user-defined tool of funasr and successfully made it working . Could you share your schema of this tool of funasr?
